### DISCUSSION TITLE
Cannot convert value "System.String" to type "System.Boolean"

### DISCUSSION THREAD

**Discussion by JaeEmAre67** [pester/Pester#2086] [Q&A] [answered]
Brand new to Pester.

`Describe "Get timestamp for output file" {`
`    It "returns <expected> (param)" -TestCases @(`
`        @{ Param = "$true $true"; Expected = Get-Date -Format "yyyyMMddHHmm" }`
`        @{ Param = "$true $false"; Expected = Get-Date -Format "yyyyMMdd" }`
`        @{ Param = "$false $true"; Expected = Get-Date -Format "HHmm" }`
`        @{ Param = "$false $false"; Expected = Get-Date -Format "yyyyMMdd" }`
`        @{ Param = ""; Expected = Get-Date -Format "yyyyMMdd" }`
`    ) {`
`        get-util-timeStamp $param | Should -Be $expected;`
`    }`
`}`

In the errors I can see it is indeed returning the date in the specified format, however all are failing with the error:
PSInvalidCastException: Cannot convert value "System.String" to type "System.Boolean". Boolean parameters accept only Boolean values and numbers, such as $True, $False, 
1 or 0.
 ArgumentTransformationMetadataException: Cannot convert value "System.String" to type "System.Boolean". Boolean parameters accept only Boolean values and numbers, such as $True, $False, 1 or 0.
 ParameterBindingArgumentTransformationException: Cannot process argument transformation on parameter 'date'. Cannot convert value "System.String" to type "System.Boolean". Boolean parameters accept only Boolean values and numbers, such as $True, $False, 
1 or 0.
 at <ScriptBlock>, c:\myapp\tests\utils.Tests.ps1:13

Which would make a bit of sense to me *if* it wasn't returning the expected value
`[-] Get timestamp for output file.returns 202109201037 22ms (18ms|4ms)`
`[-] Get timestamp for output file.returns 20210920 3ms (2ms|1ms)`
`[-] Get timestamp for output file.returns 1037 2ms (1ms|1ms)`
`[-] Get timestamp for output file.returns 20210920 2ms (2ms|1ms)`
`[-] Get timestamp for output file.returns 20210920 2ms (1ms|1ms)`

Can anyone please advise where I'm wrong?

**Comment by fflaten** [comment, marked as answer]
It's not Pester that's complaining about `Expected`, it is your function complaining about the input. Your function probably expects one or two bool-parameters but you give it a single string. 

You're effectively running `get-util-timeStamp "True True"`. "True True" can't be converted to bool
